src/core/execute.c

```diff
@@ -7092,6 +7101,29 @@ int exec_directory_add(ExecDirectory *d, const char *path, const char *symlink)
         return 1; /* new item is added */
 }
 
+static int exec_directory_item_compare_func(const ExecDirectoryItem *a, const ExecDirectoryItem *b) {
+        assert(a);
+        assert(b);
+
+        return path_compare(a->path, b->path);
+}
+
+void exec_directory_sort(ExecDirectory *d) {
+        assert(d);
+
+        if (d->n_items <= 1)
+                return;
+
+        typesafe_qsort(d->items, d->n_items, exec_directory_item_compare_func);
+
+        for (size_t i = 1; i < d->n_items; i++)
+                for (size_t j = 0; j < i; j++)
+                        if (path_startswith(d->items[i].path, d->items[j].path)) {
```

Comment:
hmm, so you are checking any of the earlier items are a prefix of the current one, iiuc? but why does that matter for symlink creation? not following...

Reply:
See the above comment.